modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/segstore/IndexFileManager.java

```diff
@@ -120,20 +129,36 @@ class IndexFileManager {
         Files.createDirectories(indexFilesDir);
     }
 
+    void start() throws IOException {
+        try (Stream<Path> indexFiles = Files.list(indexFilesDir)) {
+            Iterator<Path> it = indexFiles.sorted().iterator();
+
+            while (it.hasNext()) {
+                recoverIndexFileMetas(it.next());
+            }
```

Comment:
Just a question: why did you mix iterators and streams here? A simple `forEach` would do the job just as well, is there some important difference that I'm unaware of?

Reply:
because of checked exceptions